src/app/layout.tsx

```diff
@@ -52,33 +51,45 @@ if ((!themeChosen && isDarkModePreferred) || themeChosen === "dark") {
           name="twitter:image"
           content="https://www.rustcrab.com/rust_lgo_720.png"
         />
-        <Script
-          async
-          src="https://www.googletagmanager.com/gtag/js?id=G-J6GRE0TKHY"
-        ></Script>
-        <Script id="google-analytics">
-          {`
-            window.dataLayer = window.dataLayer || [];
-            function gtag(){dataLayer.push(arguments);}
-            gtag('js', new Date());
-            gtag('config', 'G-J6GRE0TKHY');
-          `}
-        </Script>
-      </head>
-      <Head>
-        <title>Rustcrab - The Non-Crap Repo for Rust Developers</title>
-        <meta
-          name="description"
-          content="The best repository for Rust developers"
-        />
+
+        {process.env.NODE_ENV === "production" && (
+          <>
+            <Script
+              async
+              src="https://www.googletagmanager.com/gtag/js?id=G-J6GRE0TKHY"
+            ></Script>
+
+            <Script id="google-analytics">
+              {`
+                window.dataLayer = window.dataLayer || [];
+                function gtag(){dataLayer.push(arguments);}
+                gtag('js', new Date());
+                gtag('config', 'G-J6GRE0TKHY');
+              `}
+            </Script>
+          </>
+        )}
+
         <link rel="icon" href="/favicon.ico" />
-      </Head>
-      <body className={`${inter.className} ${roboto.className} bg-white dark:bg-black min-h-screen text-black dark:text-white `}>
+      </head>
+
+      <body
+        className={`${inter.className} ${roboto.className} bg-white dark:bg-black min-h-screen text-black dark:text-white `}
+      >
         <Header />
+
         {children}
+
         <Footer />
+
+        <CookieConsent />
```

Comment:
This component was in the wrong location, producing a DOM error. It needed to be inside the `<body>`.